presto-iceberg/src/main/java/com/facebook/presto/iceberg/RuntimeStatsMetricsReporter.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.common.RuntimeStats;
+import com.facebook.presto.common.RuntimeUnit;
+import org.apache.iceberg.metrics.MetricsReport;
+import org.apache.iceberg.metrics.MetricsReporter;
+import org.apache.iceberg.metrics.ScanReport;
+
+/**
+ * A MetricsReporter implementation for reporting
+ * Iceberg scan metrics to Presto's RuntimeStats.
+ */
+
+public final class RuntimeStatsMetricsReporter
+        implements MetricsReporter
+{
+    /**
+     * RuntimeStats variable used for storing scan metrics from Iceberg reports.
+     */
+    private final RuntimeStats runtimeStats;
+
+    /**
+     * Constructs a RuntimeStatsMetricsReporter.
+     *
+     * @param runtimeStat the RuntimeStats instance to report metrics to
+     */
+    public RuntimeStatsMetricsReporter(final RuntimeStats runtimeStat)
+    {
+        this.runtimeStats = runtimeStat;
+    }
+
+    /**
+     * Helper method to construct the full metric name for a table scan.
+     *
+     * @param tableName  the name of the table
+     * @param metricName the name of the metric
+     * @return the composed metric name in the format: table.scan.metric
+     */
+    private String tableScanString(final String tableName, final String metricName)
+    {
+        return tableName + ".scan." + metricName;
+    }
+
+    @Override
+    public void report(final MetricsReport report)
+    {
+        if (!(report instanceof ScanReport)) {
+            return;
+        }
+
+        // Cast report to scanReport instance
```

Comment:
Let's remove comments that don't really add value. It's pretty easy to tell from reading the code we're doing a cast